### PR TITLE
cyclonedx-python: init at 0.4.3

### PR DESCRIPTION
**Pull request by cole-h**
###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 120002` at c0e3f61e run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>3 packages built successfully:</summary>
  <ul>
    <li>cyclonedx-python</li>
    <li>python38Packages.requirements-parser</li>
    <li>python39Packages.requirements-parser</li>
  </ul>
</details>
<details>
  <summary>2 suggestions:</summary>
  <ul>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/no-python-tests.md">no-python-tests</a>

Test runner could not discover any test cases: ‘Ran 0 tests in 0.000s’
Near pkgs/tools/misc/cyclonedx-python/default.nix:36:0:
```
   |
36 |     description = "Creates CycloneDX Software Bill of Materials (SBOM) from Python projects";
   | ^
```
</li>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/no-python-tests.md">no-python-tests</a>

Test runner could not discover any test cases: ‘Ran 0 tests in 0.008s’
Near pkgs/development/python-modules/requirements-parser/default.nix:20:0:
```
   |
20 |     description = "A Pip requirements file parser";
   | ^
```
</li>
  </ul>
</details>

File: pkgs/development/python-modules/requirements-parser/default.nix [hunk 1]
```diff
@@ -0,0 +1,26 @@
+{ lib
+, buildPythonPackage
+, fetchPypi
+, nose
+}:
+buildPythonPackage rec {
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/requirements-parser/default.nix:6]:
```suggestion
}:

buildPythonPackage rec {
```

File: pkgs/development/python-modules/requirements-parser/default.nix [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+  pname = "requirements-parser";
+  version = "0.2.0";
+
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "5963ee895c2d05ae9f58d3fc641082fb38021618979d6a152b6b1398bd7d4ed4";
+  };
+
+  propagatedBuildInputs = [
+    nose
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/requirements-parser/default.nix:16]: You are sure this is not a check only dep?

File: pkgs/development/python-modules/requirements-parser/default.nix [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+  ];
+
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/requirements-parser/default.nix:18]: Missing checkPhase and pythonImportsCheck.

File: pkgs/development/python-modules/requirements-parser/default.nix [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+  meta = with lib; {
+    description = "A Pip requirements file parser";
+    homepage = "https://github.com/davidfischer/requirements-parser";
+    license = licenses.bsd2;
+    maintainers = teams.determinatesystems.members;
+  };
+}
+
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/requirements-parser/default.nix:26]: One final new line to much.

File: pkgs/tools/misc/cyclonedx-python/default.nix [hunk 5]
```diff
@@ -0,0 +1,42 @@
+{ lib
+, python3
+, fetchFromGitHub
+}:
+python3.pkgs.buildPythonApplication rec {
```
Comment (SuperSandro2000) [inline pkgs/tools/misc/cyclonedx-python/default.nix:5]:
```suggestion
}:

python3.pkgs.buildPythonApplication rec {
```

File: pkgs/tools/misc/cyclonedx-python/default.nix [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+  ];
+
+  meta = with lib; {
+    description = "Creates CycloneDX Software Bill of Materials (SBOM) from Python projects";
+    homepage = "https://github.com/CycloneDX/cyclonedx-python";
+    license = licenses.asl20;
+    maintainers = teams.determinatesystems.members;
+  };
+}
+
```
Comment (SuperSandro2000) [inline pkgs/tools/misc/cyclonedx-python/default.nix:42]: One final new line to much.

File: pkgs/tools/misc/cyclonedx-python/default.nix [hunk 7]
```diff
@@ -0,0 +1,42 @@
...
+    requests
+    xmlschema
+    setuptools
+    requirements-parser
+    packaging
+    chardet
+    pytest
+    jsonschema
+  ];
+
```
Comment (SuperSandro2000) [inline pkgs/tools/misc/cyclonedx-python/default.nix:34]: Does it have tests?

```suggestion

  pythonImportsCheck = [ "cyclonedx" ];

```

**Pull request merged**